net/message/relay_message.go

```diff
@@ -0,0 +1,65 @@
+package message
+
+import (
+	"fmt"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/athanorlabs/atomic-swap/common/vjson"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+)
+
+// RelayClaimRequest implements common.Message for our p2p relay claim requests
+type RelayClaimRequest struct {
+	SFContractAddress ethcommon.Address          `json:"sfContractAddress" validate:"required"`
+	Swap              *contracts.SwapFactorySwap `json:"swap" validate:"required"`
+	Secret            []byte                     `json:"secret" validate:"required,len=32"`
```

Comment:
what's the difference between this and having the type be `[32]byte`?

Reply:
With `[32]byte`, go will use the annoyingly verbose `{1, 128, 240, 10, ...}` JSON encoding of arrays. If we created a custom type for `[32]byte`, I could marshal it as base64 or base16, but since we didn't have a custom type, using a byte slice with size validation seemed like an easy solution to get base64 encoding.